pkg/collector/corechecks/snmp/snmp.go

```diff
@@ -161,30 +60,27 @@ func (c *Check) Configure(rawInstance integration.Data, rawInitConfig integratio
 		return fmt.Errorf("common configure failed: %s", err)
 	}
 
-	config, err := buildConfig(rawInstance, rawInitConfig)
+	c.config, err = checkconfig.NewCheckConfig(rawInstance, rawInitConfig)
 	if err != nil {
 		return fmt.Errorf("build config failed: %s", err)
 	}
+	log.Debugf("SNMP configuration: %s", c.config.ToString())
 
-	log.Debugf("SNMP configuration: %s", config.toString())
-
-	c.config = config
-	err = c.session.Configure(c.config)
+	c.singleDeviceCk, err = devicecheck.NewDeviceCheck(c.config, c.config.IPAddress)
```

Comment:
Why are you passing both `c.config` and `c.config.IPAddress` ? By passing `c.config` the callee should have all the info.

Reply:
@pducolin That's because `devicecheck.NewDeviceCheck` will be used for Autodiscovery, and in that case, we will have different IPAddress: https://github.com/DataDog/datadog-agent/pull/8996/files#diff-a018d0e4a2a347a12b1e22be938ccd9eaef390c0e6aeedb89ad821a6c66f238dR205